js/src/TooltipContainer.js

```diff
@@ -83,30 +114,57 @@ function is_visible () {
 
 /**
  * Show and place the input.
- * @param {String} type - 'reaction_label', 'node_label', or 'gene_label'
+ * @param {string} type - 'reaction_label', 'node_label', or 'gene_label'
  * @param {Object} d - D3 data for DOM element
  * @param {Object} coords - Object with x and y coords. Cannot use coords from
  *                          'd' because gene labels do not have them.
  */
 function show (type, d) {
   // get rid of a lingering delayed hide
-  this.cancel_hide_tooltip()
+  this.cancelHideTooltip()
 
   if (_.contains([ 'reaction_label', 'node_label', 'gene_label' ], type)) {
-    var coords = { x: d.label_x, y: d.label_y + 10 }
+    // Use a default height if the ref hasn't been connected yet
+    const tooltipSize = this.tooltipRef !== null && this.tooltipRef.getSize
+      ? this.tooltipRef.getSize()
+      : { width: 270, height: 100 }
+    this.currentTooltip = { type, id: d[type.replace('_label', '_id')] }
+    var windowTranslate = this.zoomContainer.window_translate
+    var windowScale = this.zoomContainer.window_scale
+    var mapSize = this.map !== null ? this.map.get_size() : { width: 1000, height: 1000 }
+    if (this.getTooltipSize !== null) {
+      //  console.log(this.getTooltipSize())
+    }
+    var offset = {x: 0, y: 0}
+    if (mapSize.width < 500) {
```

Comment:
Isn't it possible to tweak coefficients/constants with this conditional and make the actual computations more simple?

Reply:
What did you have in mind?